arcade/sound.py

```diff
@@ -44,6 +44,9 @@ def __init__(self, file_name: Union[str, Path], streaming: bool = False):
             self.file_name, streaming=streaming
         )
 
+        if self.source.duration is None:
+            raise Exception("Audio duration must be known when loaded, but this audio source returned `None`")
```

Comment:
This is probably better as a `ValueError` or a more specific type. I'm not sure if pyglet provides media-related exceptions, but we could probably add a `ValueError` subclass to `arcade.utils` or wherever the `ValueError` subclasses like `ByteRangeError` are right now.

Reply:
What should the exception be called?  The docs say `ValueError` means the wrong kind of value was passed, such as a number out of range.  Is that what triggers a source to have no duration? I don't really know.

https://docs.python.org/3/library/exceptions.html#ValueError

Reply:
> Is that what triggers a source to have no duration? I don't really know.

At least two backends supported by pyglet can fail to know the duration of an audio / video source:

* https://superuser.com/questions/1179000/ffmpeg-get-duration-of-video-file-without-meta-data
* https://github.com/beetbox/audioread/issues/31

There are also might be unlikely edge cases like trying to read from a live streaming audio / video source for which the length can't be known beforehand.

Although `ValueError` seems the most appropriate base exception type since we're receiving a value which doesn't match the expected requirements, there are also [some exception types in pyglet which may be relevant](https://github.com/pyglet/pyglet/blob/ff89b4f68db321a2e1699a8e70ca7fcf06d52ae0/pyglet/media/exceptions.py):

* `MediaException(Exception)`
* `MediaFormatException(MediaException)`
* `CannotSeekException(MediaException)`

`MediaException` seems like the most appropriate of them, but I'm not sure either.